src/otaclient_common/downloader.py

```diff
@@ -238,7 +237,7 @@ def retry_on_digest_mismatch(func: Callable[P, T]) -> Callable[P, T]:
     def _wrapper(*args: P.args, **kwargs: P.kwargs) -> T:
         try:
             return func(*args, **kwargs)
-        except (HashVerificationError, BrokenDecompressionError) as e:
+        except (PartialDownload, HashVerificationError, BrokenDecompressionError) as e:
```

Comment:
(nits) How about to catch all of the inherited exceptions from `DownloadError`?
We don't have to care about similar issues in future.
```suggestion
        except DownloadError as e:
```

Reply:
IMO, being explicit is better here, so I prefer to list the concrete exception types here.